blog/2025-08-26-tigris-veo/index.mdx

```diff
@@ -0,0 +1,383 @@
+const S3_ENDPOINT = "https://t3.storage.dev";
+const S3_FORCE_PATH_STYLE = false;
+
+const s3 = new S3Client({
+  region: S3_REGION,
+  endpoint: S3_ENDPOINT,
+  forcePathStyle: S3_FORCE_PATH_STYLE,
+  credentials: {
+    accessKeyId: process.env.TIGRIS_STORAGE_ACCESS_KEY_ID!,
+    secretAccessKey: process.env.TIGRIS_STORAGE_SECRET_ACCESS_KEY!,
+  },
+});
+
+async function uploadObject(
+  buffer: Buffer,
+  contentType: string,
+  objectKey: string
+) {
+  await s3.send(
+    new PutObjectCommand({
+      Bucket: S3_BUCKET_NAME,
+      Key: objectKey,
+      Body: buffer,
+      ContentType: contentType,
+    })
+  );
+}
+
+async function getObject(bucket: string, objectKey: string) {
+  const command = new GetObjectCommand({
+    Bucket: bucket,
+    Key: objectKey,
+  });
+  return await getSignedUrl(s3, command, { expiresIn: 60 * 60 });
+}
+```
+
+In the code above, we created an S3 client. Tigris is S3 compatible and supports
+S3 SDK and CLI. We also added two functions (`uploadObject` and `getObject`) to
+upload the videos to our bucket, and then read the video from it.
+
+Use `uploadObject` and `getObject` in `generateVideoFromText`:
+
+```typescript
+async function generateVideoFromText(
+  prompt: string,
+  numberOfVideos = 1,
+): Promise<string[]> {
+  …
+
+  if (operation?.response) {
+    const videos = operation.response?.generatedVideos;
+    if (videos === undefined || videos.length === 0) {
+      throw new Error('No videos generated');
+    }
+
+    return await Promise.all(
+      videos.map(async (generatedVideo: GeneratedVideo, index: number) => {
+        const uri = `${generatedVideo.video?.uri || ''}?key=${API_KEY}`;
+        const res = await fetch(uri);
+
+        if (!res.ok) {
+          throw new Error(`Failed to fetch video: ${res.status} ${res.statusText}`);
+        }
+        const contentType = res.headers.get('content-type') || 'application/octet-stream';
+        const arrayBuffer = await res.arrayBuffer();
+        const buffer = Buffer.from(arrayBuffer);
+
+        const extension = contentType.includes('mp4') ? '.mp4' : contentType.includes('quicktime') ? '.mov' : '';
+        const uniqueId = (global as any).crypto?.randomUUID?.() || `${Date.now()}-${Math.random().toString(36).slice(2)}`;
+        const objectKey = `videos/${uniqueId}${extension || `.mp4`}`;
+
+        await uploadObject(buffer, contentType, objectKey);
+
+        const videoUrl = await getObject(S3_BUCKET_NAME, objectKey);
+        return videoUrl;
+      }),
+    );
+  } else {
+    throw new Error('No videos generated');
+  }
+}
+```
+
+## Enhance the Veo3 prompt
+
+<video controls width="100%">
+  <source
+    src="/blog/img/blog/veo3-tigris/enhance-prompt.mp4"
+    type="video/mp4"
+  />
+  Download the <a href="/blog/img/blog/veo3-tigris/enhance-prompt">MP4</a> version.
+</video>
+
+With text‑to‑video working and objects stored on Tigris, let’s boost prompt
+quality—and video quality—by structuring the inputs.
```

Comment:
```suggestion
quality and video quality by structuring the inputs.
```

-Wpedantic: the em dash isn't needed here, the intent is that "making the prompt better makes the output better", so the "and video quality" bit flows better inline to the sentence.